sdk/go/pulumi/context.go

```diff
@@ -806,6 +806,20 @@ func (ctx *Context) registerResource(
 	parent := options.Parent
 	if options.Parent == nil {
 		options.Parent = ctx.stack
+	} else {
+		// Guard against uninitialized parent resources to prevent
+		// panics from invalid state further down the line.
+		// Uninitialized parent resources won't have a URN.
+		if parent.URN().getState() == nil {
+			var msg string
+			if _, parentIsCustom := parent.(CustomResource); !parentIsCustom {
+				msg = "parent component resource %T has not been registered: " +
+					"did you mean to call RegisterComponentResource?"
```

Comment:
"did you mean to call" sounds odd, they _must_ have meant to do that, there's nothing else sensible to do with a component resource.

Reply:
Fair. How about "did you forget to call"?

Reply:
Again I don't think there's any "did" about it, @susanev probably has words to say on phrasing error messages but I think this should probably be something like "parent component resource %T has not been registered: RegisterComponentResource should have been called"

Reply:
I think we can drop the two sentence problem completely with:

    parent component resource %T has not been registered via RegisterComponentResource

Thanks to @ericrudder for the suggestion